### PR TITLE
Add post body to har requests

### PR DESCRIPTION
**Pull request by dachrillz**
probably merge #12 first

### CODE REVIEW

**Comment by dachrillz** [comment]
add proper postData to har requests
Now post data is added to incoming requests. This paves the way for
veriyfing their content against openapispec in future commits.

File: src/httpParsing/parseHar.ts [hunk 1]
```diff
@@ -11,7 +11,27 @@ const getEntries = (harFile: Record<string, any>) => {
...
+    return undefined;
+  }
+  if (input.mimeType !== "application/json") {
+    console.warn("Only application/json is supported!");
+    return undefined;
+  }
+  return {
+    mimeType: input.mimeType,
+    text: input.text,
+    parsed: JSON.parse(input.text),
```
Comment (bittrance) [inline src/httpParsing/parseHar.ts]: I think parsing the content here will 1) be memory-intensive and 2) make adding additional mime types tricky. I think we should keep the string and only parse it if and when we validate it.

File: src/httpParsing/model.ts [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+
+export type JSONObject = {
+  [x: string]: JSONValue;
+};
+
+export type JSONArray = Array<JSONValue>;
+
+export type postData = {
+  mimeType: string;
+  parsed?: JSONObject | JSONArray;
```
Comment (bittrance) [inline src/httpParsing/model.ts]: I don't think we should do parsing here, but also, when you talk about JSON, by definition, this is `any`.

**Review by bittrance — CHANGES_REQUESTED** [review]
I think we should delay parsing till we know that we want to validate it.

File: src/httpParsing/parseHar.ts (hunk 1 above)
Comment (dachrillz) [reply src/httpParsing/parseHar.ts]: agreed, will change

File: src/httpParsing/model.ts (hunk 2 above)
Comment (dachrillz) [reply src/httpParsing/model.ts]: agreed, removed parsing from this pr

File: src/httpParsing/parseHar.ts [hunk 3]
```diff
@@ -11,7 +11,27 @@ const getEntries = (harFile: Record<string, any>) => {
   return entries;
 };
 
+const parseContent = (input: {
```
Comment (bittrance) [inline src/httpParsing/parseHar.ts]: Nit-picking perhaps, but this method doesn't actually parse content. It seems more like tryFormContenNode or something.

**Review by bittrance — APPROVED** [review]

**Pull request merged**